packages/apollos-ui-auth/src/AuthSMS/PhoneEntry.js

```diff
@@ -0,0 +1,165 @@
+    smsPolicyInfo: (
+      <H6>
+        {"We'll never share your information or contact you (unless you ask!)."}
+      </H6>
+    ),
+    allowPassword: true,
+    smsPasswordLoginPrompt: "I'd rather use my email and a password",
+  };
+
+  validationSchema = Yup.object().shape({
+    phone: Yup.string().matches(/^\D?(\d{3})\D?\D?(\d{3})\D?(\d{4})$/),
+  });
+
+  get flatProps() {
+    return { ...this.props, ...(this.props.screenProps || {}) };
+  }
+
+  submitHandler = (mutate) => async (
+    { phone },
+    { setSubmitting, setFieldError }
+  ) => {
+    setSubmitting(true);
+    try {
+      await mutate({ variables: { phone } });
+      this.props.navigation.navigate('AuthSMSVerification', { phone });
+    } catch (e) {
+      setFieldError(
+        'phone',
+        'There was an error. Please double check your number and try again.'
+      );
+    }
+    setSubmitting(false);
+  };
+
+  handlePasswordLoginPress = () => {
+    this.props.navigation.navigate('AuthPassword');
+  };
+
+  render() {
+    const {
+      brand,
+      authTitleText,
+      smsPromptText,
+      smsPolicyInfo,
+      allowPassword,
+      smsPasswordLoginPrompt,
+    } = this.flatProps;
+
+    return (
+      <KeyboardAvoidingView style={StyleSheet.absoluteFill} behavior="padding">
+        <BackgroundView>
+          <Mutation mutation={requestPin}>
+            {(mutate) => (
+              <Formik
+                initialValues={{ phone: '' }}
+                validationSchema={this.validationSchema}
+                onSubmit={this.submitHandler(mutate)}
+              >
+                {({
+                  handleChange,
+                  handleSubmit,
+                  values,
+                  isSubmitting,
+                  isValid,
+                  touched,
+                  errors,
+                }) => (
+                  <SafeAreaView style={StyleSheet.absoluteFill}>
+                    <ScrollView>
+                      <PaddedView>
+                        {brand}
+                        <TitleText>{authTitleText}</TitleText>
+                        <PromptText padded>{smsPromptText}</PromptText>
+
+                        <TextInput
+                          autoFocus
+                          autoComplete="tel"
+                          label="Mobile Number"
+                          type="phone"
+                          returnKeyType="next"
+                          onSubmitEditing={this.handleAdvance}
+                          enzblesReturnKeyAutomatically
+                          error={touched.phone && errors.phone}
+                          onChangeText={handleChange('phone')}
+                          value={values.phone}
+                        />
+                      </PaddedView>
+                      {allowPassword ? (
+                        <PaddedView>
+                          <ButtonLink onPress={this.handlePasswordLoginPress}>
```

Comment:
If my suggestion above is accepted you'll want this to match 😉 
```suggestion
                          <ButtonLink onPress={this.handleOnPressLogin}>
```